crawl-ref/source/skills.cc

```diff
@@ -1160,7 +1161,7 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
     {
         // each loop is the max skill points that can be gained at the
         // current cost level, up to `target`.
-        int next_level = reduced ? skill_cost_needed(you_skill_cost_level) :
+        const int next_level = reduced ? skill_cost_needed(you_skill_cost_level) :
```

Comment:
What about something like:

    const int target_level  = you_skill_cost_level + (reduced ? 0 : 1);
    const int next_level = skill_cost_needed(target_level);

Factors out the repeated `skill_cost_needed` and `you_skill_cost_level`, making it clearer what the impact of `reduced` is. Not sure about the name for my suggested intermediate variable, since I'm not totally sure what you're computing here.